normalize seed scope to always 'wwwN.' + add docs on URL Normalization (#1088)
webrecorder/browsertrix-crawler: docs/docs/user-guide/crawl-scope.md

- `custom` — crawl based on the `--include` regular expression rules.

> Note that the scope settings for multi-page crawls (page-spa, prefix, host, domain) are also affected by [URL Normalization](#url-normalization)

Reviewer: ```suggestion
Note that the scope settings for multi-page crawls (page-spa, prefix, host, domain) are also affected by [URL Normalization](#url-normalization).
```

The `--extraHops` setting can be set globally or per seed to allow expanding the current inclusion scope N 'hops' beyond the configured scope. Note that this mechanism only expands the inclusion scope, and any exclusion rules are still applied. If a URL is to be excluded via the exclusion rules, that will take precedence over the `--extraHops`.

## URL Normalization

The crawler applies 'URL normalization' rules in a few places, 'normalizing' the URL to a slightly different
format to treat similar URLs as the same and ignore insignificant differences.

- For `prefix`, `host` and `domain` scope types, the `www<optional number>.` prefix and `http` or `https` schemes
are ignored. For example, given a seed with `prefix` scope of `http://example.com/path/`, URLs starting with `https://example.com/path/` and `https://www.example.com/path/` will also be included.

- For all pages, the query arguments are sorted, so that URL `https://example.com/query?A=1&B=2` and `https://example.com/query?B=2&A=1` are considered the same URL and crawled only once.

## Seed Page Redirects

> Given the URL normalization rules above, if a seed pages redirect to another page, eg. if `https://example.com/` redirects to `http://example.com/` or `http://www.example.com/`, they are treated as the same seed, and no new seed is added.

Reviewer: ```suggestion
Given the URL normalization rules above, if a seed page redirects to another page that differs only by scheme or www prefix, e.g. if `https://example.com/` redirects to `http://example.com/` or `http://www.example.com/`, they are treated as the same seed, and no new seed is added.
```

> If the seed page redirects to another URL, eg. `http://example.com/` to `https://example.org`, the new page is *not*

Author: ```suggestion
If the seed page redirects to another URL, eg. `http://example.com/` to `https://example.org/`, the new page is *not*
```

> If the seed page redirects to another URL, eg. `http://example.com/` to `https://example.org/`, the new page is *not*

Reviewer: ```suggestion
If the seed page redirects to another URL, e.g. `http://example.com/` to `https://example.org/`, the new page is *not*
```

> added as a new seed by default, but that immediate page is crawled (unless the page was explicitly excluded)

Reviewer: ```suggestion
If the seed page redirects to another URL, e.g. `http://example.com/` to `https://example.org`, the new page is *not*
added as a new seed by default, but that immediate page is crawled (unless the page was explicitly excluded).
```

> added as a new seed by default, but that immediate page is crawled (unless the page was explicitly excluded)

Author: ```suggestion
added as a new seed by default, but that immediate page is crawled (unless the page was explicitly excluded).
```

> The `--addRedirectedSeeds` can be used to always add the new page URL as a seed, with the same scope as the original seed.

Reviewer: ```suggestion
The `--addRedirectedSeeds` flag can be used to always add the new redirected page URL as a seed, with the same scope as the original seed.
```

> Note that this can lead to unintended consequences, for example if the seed is `https://example.com/path/` and it redirects to `https://example.org/` with `scopeType: prefix`, the new seed added is `https://example.org/` with `scopeType: prefix` as well, but potentially expanding the size of the crawl.

Reviewer: ```suggestion
Note that this can lead to unintended consequences. For example, if the seed is `https://example.com/path/` and it redirects to `https://example.org/` with `scopeType: prefix`, the new seed added is `https://example.org/` with `scopeType: prefix` as well, potentially expanding the size of the crawl.
```

> Note that this can lead to unintended consequences, for example if the seed is `https://example.com/path/` and it redirects to `https://example.org/` with `scopeType: prefix`, the new seed added is `https://example.org/` with `scopeType: prefix` as well, but potentially expanding the size of the crawl.

Author: ```suggestion
Note that this can lead to potentially broader or narrower scope, for example if the seed is `https://example.com/path/` and it redirects to `https://example.org/` with `scopeType: prefix`, the new seed added is `https://example.org/` with `scopeType: prefix` as well, but potentially expanding the size of the crawl.
```

> Note that this can lead to potentially broader or narrower scope, for example if the seed is `https://example.com/path/` and it redirects to `https://example.org/` with `scopeType: prefix`, the new seed added is `https://example.org/` with `scopeType: prefix` as well, but potentially expanding the size of the crawl.

Author: ```suggestion
Note that this can lead to potentially broader or narrower scope. For example, if the seed is `https://example.com/path/` and it redirects to `https://example.org/` with `scopeType: prefix`, the new seed added is `https://example.org/` with `scopeType: prefix` as well, but potentially expanding the size of the crawl.
```

!!! note "Previous behavior"

>     Prior to 1.14.0 release, crawler would always add all redirected seeds as new seeds. The equivalent behavior can now be enabled with the `--addRedirectedSeeds` flag.

Reviewer: ```suggestion
    Prior to the 1.14.0 release, the crawler would always add all redirected seeds as new seeds. The equivalent behavior can now be enabled with the `--addRedirectedSeeds` flag.
```
